modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,10 +48,10 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybePriorityPeerIds: Option[NonEmptySet[PeerId]] = none
```

Comment:
what about exposing a method on the storage like `getPeers` - `getPriorityPeers`? that way you keep the implementation self enclosed within the storage how to retrieve the priority peers

Reply:
@marcinwadon Another thought: If this code is supposed to be ripped out later is it worth it to put it in the storage class?
cc @ryle-goehausen

Reply:
I'm not sure what you're suggesting or what we'd gain by doing that.
`prioritySeedList` -- like `seedlist` -- are optional values received by the CLI.

Reply:
so, my suggestion was, to extend the cluster storage by additional method:
```scala
def getPriorityPeers
```
so that you have a single place where you implement how to obtain the priority peers.
ideally, it should be something like:
```scala
def getPriorityPeers = getPeers.map(_.filter(_.isPriorityPeer))
```

to achieve that, instead of adding a new seedlist (named priority seedlist) which adds additional overhead (management of the seedlist), you may add a single field `isPriorityPeer` to the existing seedlist, to mark peer as priority. that should be very easy since we are able to extend the seedlist by additional fields thanks to @cngo-github change.

whenever you would like to use these peers, you can do (or something similar):
```scala
if (realignWithNonPrioritySeedlist) 
  clusterStorage.getPriorityPeers 
else
  clusterStorage.getPeers
```

Reply:
@marcinwadon I will ask @ryle-goehausen about storing the priority peers in `L0ClusterStorage`.
However, your implementation above is not equivalent to what we have now. AFAIK the peers on the either seedlist are fixed at startup whereas `clusterStorage.getPeers` will not necessarily return the seedlist and can change over time. I don't think that's the intention of the using the seedlist in the algorithm.

Reply:
Yeah, I'm a little confused here since we're not intending to ever change or update the storage internally. That should make it a fixed configuration here.

Reply:
@sdavidp @ryle-goehausen ok, so you want to query nodes that are not a part of the network. if so, then it shouldn't be in the cluster storage indeed

Reply:
See other discussion point regarding not using IPs yet but rather relying on internal storage per @TheMMaciek suggestion -- since we can assume to rely on peers already populated we don't need to modify the internal (only keep track of the embedded ones)